### ISSUE TITLE
Better array handling

### ISSUE THREAD

**Issue by m-mohr** [Open-EO/openeo-processes#114]
We have some cases where we may want to iterate over an array so I'm thinking about either defining a function array_each or allow arrays as input data for apply.

array_each (the definition is a bit outdated):
```json
{
    "id": "array_each",
    "summary": "Applies a unary process to each array element",
    "description": "Applies a **unary** process which takes a single value such as `abs` or `sqrt` to each value in the array.",
    "categories": [
        "arrays"
    ],
    "parameter_order": ["data", "process"],
    "parameters": {
        "data": {
            "description": "An array.",
            "schema": {
                "type": "array",
                "items": {
                    "description": "Any data type is allowed."
                }
            },
            "required": true
        },
        "process": {
            "description": "A process (callback) to be applied on each value. The specified process must be unary meaning that it must work on a single value.",
            "schema": {
                "type": "object",
                "format": "callback",
                "parameters": {
                    "x": {
                        "description": "A value of any data type could be passed."
                    }
                }
            },
            "required": true
        }
    },
    "returns": {
        "description": "An array with the newly computed values. The number of elements are the same as for the original array.",
        "schema": {
			"type": "array",
			"items": {
				"description": "Any data type is allowed."
			}
        }
    }
}
```

**Comment by m-mohr** [comment]
It's probably also useful to define array_filter and array_reduce?!

**Comment by m-mohr** [comment]
Solved with PR #121.
